src/main/java/org/babi/backend/common/dao/DaoUtil.java

```diff
@@ -1,28 +1,21 @@
 package org.babi.backend.common.dao;
 
-import java.util.List;
-import java.util.Optional;
+import java.util.Set;
 
 public class DaoUtil {
 
-    public static String buildValuesTuple(List<List<Object>> values) {
-        final StringBuilder sql = new StringBuilder();
-        Optional.ofNullable(values)
-                .ifPresent(v -> v.forEach(tuple -> {
-                    sql.append("(");
-                    tuple.forEach(t -> {
-                        sql.append(String.format("%s, ", t));
-                    });
-                    final int length = sql.length();
-                    sql.replace(length - 2, length, "), ");
-                }));
+    public static String buildUnlinkQuery(String table, Set<Long> ids, String identifierColumn, String identifierBind, String nestedColumn) {
+        final StringBuilder sql = new StringBuilder(String.format("delete from %s where %s = :%s and %s in (", table, identifierColumn, identifierBind, nestedColumn));
+        ids.forEach(id -> sql.append(String.format("%d, ", id)));
         final int length = sql.length();
+        return sql.replace(length - 2, length, ")").toString();
+    }
 
-        if (length >= 2) {
-            sql.replace(length - 2, length, "");
-        }
-
-        return sql.toString();
+    public static String buildLinkQuery(String table, Long id, Set<Long> nestedIds, String identifierColumn, String nestedColumn) {
+        final StringBuilder sql = new StringBuilder(String.format("insert into %s(%s, %s) values ", table, identifierColumn, nestedColumn));
```

Comment:
use prepared statements to avoid sql injection

Reply:
It does not possible to make SQL injection from user's input as user can only pass the `id` and `nestedIds` params. So this method is created to prevent duplicate code for same logic but for different schemas.